### PR TITLE
Use argparse to parse CLI arguments 

### PR DESCRIPTION
**Pull request by shreyash2106**

### CODE REVIEW

File: src/autogluon_dashboard/dashboard.py [hunk 1]
```diff
@@ -24,26 +20,75 @@ def upload_to_s3(s3_client: botocore.client, file_name: str, object_name: str, a
...
+        type=str,
+        help="Location in S3 bucket to save csv file. Example: sub_folder/file_name.csv",
+        default="dev_data/autogluon.csv",
+    )
+
+    parser.add_argument(
+        "--s3_bucket",
+        type=str,
+        help="Name of S3 bucket that results to aggregate get outputted to",
+        default="dashboard-test-yash",
```
Comment (yinweisu) [inline src/autogluon_dashboard/dashboard.py:55]: Add a TODO regarding changing the default

File: src/autogluon_dashboard/dashboard.py [hunk 2]
```diff
@@ -24,26 +20,75 @@ def upload_to_s3(s3_client: botocore.client, file_name: str, object_name: str, a
...
-    s3_client = boto3.client("s3")
-    global s3_url
+    args = get_args()
+    # Set variables to corrensponding command line args
+    per_dataset_csv_path = args.per_dataset_csv
+    aggregated_csv_path = args.all_dataset_csv
+    per_dataset_s3_loc = args.per_dataset_s3
+    aggregated_s3_loc = args.all_dataset_s3
+    bucket_name = args.s3_bucket
+    s3_url = f"https://{bucket_name}.s3.us-west-2.amazonaws.com/"
```
Comment (yinweisu) [inline src/autogluon_dashboard/dashboard.py:71]: is this s3_url used for website browsing or uploading artifacts? If it's uploading artifact, I don't think you need to provide a https url but a s3 url is enough. Raise this question becuse `us-west-2` clearly doesn't work for all buckets

File: src/autogluon_dashboard/dashboard.py [hunk 3]
```diff
@@ -59,8 +104,11 @@ def run_dashboard():
             "hvplot",
         ]
     )
-    upload_to_s3(s3_client, os.path.join(web_files_dir, "out.html"), "out.html", args={"ContentType": "text/html"})
-    upload_to_s3(s3_client, os.path.join(web_files_dir, "out.js"), "out.js")
+    # Upload WebAssembly to S3 bucket
+    upload_to_s3(
+        s3_client, os.path.join(web_files_dir, "out.html"), "out.html", bucket_name, args={"ContentType": "text/html"}
+    )
+    upload_to_s3(s3_client, os.path.join(web_files_dir, "out.js"), "out.js", bucket_name)
```
Comment (yinweisu) [inline src/autogluon_dashboard/dashboard.py:112]: Probably worth logging out the web url that the user can use to view the panel

File: src/autogluon_dashboard/dashboard.py [hunk 4]
```diff
@@ -24,26 +23,103 @@ def upload_to_s3(s3_client: botocore.client, file_name: str, object_name: str, a
...
+        type=str,
+        help="Location in S3 bucket to save csv file. Example: sub_folder/file_name.csv",
+        default=CSV_FILES_DIR + "all_data.csv",  # TODO: Add commit ID (if applicable) to S3 file path
+    )
+    parser.add_argument(
+        "--all_dataset_s3",
+        type=str,
+        help="Location in S3 bucket to save csv file. Example: sub_folder/file_name.csv",
+        default=CSV_FILES_DIR + "autogluon.csv",  # TODO: Add commit ID (if applicable) to S3 file path
+    )
```
Comment (yinweisu) [inline src/autogluon_dashboard/dashboard.py:54]: Do we want the user to provide name of the csv file as well or just the sub_folder? Or are we handling user providing a path without the filename correctly?

Comment (yinweisu) [reply src/autogluon_dashboard/dashboard.py:54]: OK. So this one actually kind of conflict with `s3_prefix` you have later.
Let's make it that we only accept s3_prefix, and just save the csv files under the prefix with `utils/xxx.csv` or `dev_data/xxx.csv` like you doing right now.
Users don't need to know where the utils are stored as they should be transparent to the user

File: src/autogluon_dashboard/dashboard.py [hunk 5]
```diff
@@ -24,26 +23,103 @@ def upload_to_s3(s3_client: botocore.client, file_name: str, object_name: str, a
...
+        if region:
+            logger.warning(
+                "Cannot specify region if no bucket has been provded. Defaulting to AutoGluon bucket and region (%s)",
+                S3_REGION,
+            )
+        else:
+            logger.warning(
+                "No bucket or region has been provided. Defaulting to AutoGluon bucket and region (%s)",
+                S3_REGION,
+            )
```
Comment (yinweisu) [inline src/autogluon_dashboard/dashboard.py:89]: We don't want to give warning if user didn't specify the region. It's just the default. Users don't want to see warning when they use the default

File: src/autogluon_dashboard/dashboard.py [hunk 6]
```diff
@@ -24,26 +23,103 @@ def upload_to_s3(s3_client: botocore.client, file_name: str, object_name: str, a
...
+    else:
+        if not region:
+            raise ValueError("You must specify a region if you provide a bucket")
+        else:
+            logger.info(f"You have specified Bucket {bucket_name} and Region {region}.")
+
+    # Set s3 public urls to CSVs as environment variables
+    prefix = prefix if prefix.startswith("/") else "/" + prefix
+    prefix = prefix if prefix.endswith("/") else prefix + "/"
+    s3_url = f"https://{bucket_name}{prefix}.s3.{region}.amazonaws.com/"
```
Comment (yinweisu) [inline src/autogluon_dashboard/dashboard.py:101]: I wonder should the correct url be
`f"https://{bucket_name}s3.{region}.amazonaws.com/{prefix}/xxx.html"`?
can you verify?

Comment (shreyash2106) [reply src/autogluon_dashboard/dashboard.py:101]: Oh yeah you're right - my bad!

Comment (yinweisu) [reply src/autogluon_dashboard/dashboard.py:101]: Would recommend do an end to end test

**Review by yinweisu — APPROVED** [review]
LGTM!

**Pull request merged**